app/javascript/react/components/ReviewForm.js

```diff
@@ -0,0 +1,175 @@
+import React, { useState, useEffect } from "react";
+import { StarRating } from "@thumbtack/thumbprint-react";
+
+const ReviewForm = (props) => {
+
+  let flavorId = props.match.params.flavor_id
+
+  // These 4 states hold each of our star ratings for review attributes
+  const [overallRating, setOverallRating] = useState(null);
+  const [sweetnessRating, setSweetnessRating] = useState(null);
+  const [mouthFeelRating, setMouthFeelRating] = useState(null);
+  const [tasteRating, setTasteRating] = useState(null);
+
+  // These 4 states hold hover values for each review attribute
+  const [overallHoverRating, setOverallHoverRating] = useState(undefined);
+  const [sweetnessHoverRating, setSweetnessHoverRating] = useState(undefined);
+  const [mouthFeelHoverRating, setMouthFeelHoverRating] = useState(undefined);
+  const [tasteHoverRating, setTasteHoverRating] = useState(undefined);
+
```

Comment:
there is definitely gonna be a way we can dry this up. There should be one source of truth for state, but in this we hold something like sweetness in both reviewData state and multiple other places. the hover ratings I understand, but the overrall ratings and the reviewData info can likely be merged. See below where I talk about the callbacks

Reply:
Updated state per our discussion & your suggestion. Thanks!